packages/@aws-cdk/aws-codedeploy/lambda-packages/ecs-deployment-provider/lib/on-event.ts

```diff
@@ -0,0 +1,116 @@
+import { Logger } from '@aws-lambda-powertools/logger';
+import { CodeDeploy } from '@aws-sdk/client-codedeploy';
+
+const logger = new Logger({ serviceName: 'ecsDeploymentProviderOnEvent' });
+const codedeployClient = new CodeDeploy({});
+
+/**
+ * The properties of the CodeDeploy Deployment to create
+ */
+interface DeploymentProperties {
+  description: string;
+  applicationName: string;
+  deploymentConfigName: string;
+  deploymentGroupName: string;
+  autoRollbackConfigurationEnabled: string;
+  autoRollbackConfigurationEvents: string;
+  revisionAppSpecContent: string;
+}
+
+/**
+ * The properties in the Data object returned to CloudFormation 
+ */
+export interface DataAttributes {
+  deploymentId: string;
+}
+
+/**
+ * The request object that the custom resource lamba function receives from CloudFormation.
+ */
+export interface OnEventRequest {
+  RequestType: string;
+  LogicalResourceId: string;
+  PhysicalResourceId: string;
+  ResourceProperties: DeploymentProperties;
+  OldResourceProperties: DeploymentProperties;
+  ResourceType: string;
+  RequestId: string;
+  StackId: string;
+}
+/**
+ * The response object that the custom resource lambda function returns to CloudFormation.
+ */
+export interface OnEventResponse {
+  PhysicalResourceId?: string;
+  Data?: DataAttributes;
+  NoEcho?: boolean;
+}
+
+/**
+ * The lambda function called from CloudFormation for this custom resource.
+ * 
+ * @param event 
+ * @returns attribues of the deployment that was created
+ */
+export async function handler(event: OnEventRequest): Promise<OnEventResponse> {
+  logger.appendKeys({
+    stackEvent: event.RequestType,
+  });
+  switch (event.RequestType) {
+    case 'Create':
+    case 'Update': {
```

Comment:
What happens if we create a deployment while there is one in progress? I don't think we should just create a new one on top.

Should we look for an existing deployment and if there is one:
- throw an error to fail the stack operation?
- start tracking that one instead of creating a new one?
- something else?

Reply:
The CodeDeploy service will throw an error if a deployment is already in progress. I'd rather rely on the service to own enforcing this.